src/main/java/de/zalando/aruha/nakadi/domain/BatchItem.java

```diff
@@ -13,6 +12,11 @@
     public BatchItem(final JSONObject event) {
         this.response = new BatchItemResponse();
         this.event = event;
+
+        Optional.
+                of(event.optJSONObject("metadata"))
```

Comment:
That will throw `NullPointerException` if `event.optJSONObject("metadata")` is null.
Better use `Optional.ofNullable`.